app/workers/regenerate_image_worker.rb

```diff
@@ -0,0 +1,26 @@
+class RegenerateImageWorker
+  include Sidekiq::Worker
+
+  def perform doc_ids
+    documents = Document.find(doc_ids)
+    documents.each { |doc| doc.update! processed: false } unless documents.blank?
+    processing = false
+    current = doc_ids.pop
+    while !doc_ids.blank?
+      if processing
+        if Document.find(current).processed
+          Rails.logger.info "Regenerated images for document with id #{current},
```

Comment:
As far as I can tell, PdfToImagesWorker only operates on a single PDF per sidekiq job. So, even if one fails, it shouldn't mess up the other documents. If you're only using this in that rake task, I would loop over doc_ids there. It's ok to call PdfToImagesWorker.perform_async a thousand times and preferred over a single sidekiq job that loops over a bunch of records.